prow/cmd/deck/static/common/common.ts

```diff
@@ -169,13 +169,14 @@ export namespace tooltip {
 }
 
 export namespace icon {
-  export function create(iconString: string, tip: string = ""): HTMLAnchorElement {
+  export function create(iconString: string, tip: string = "", onClick: (this: HTMLElement, ev: MouseEvent) => any = () => null): HTMLAnchorElement {
```

Comment:
I think it'd be nicer to not add any listener if we don't need one. Consider instead writing this like this:

```ts
export function create(iconString: string, tip: string = "", onClick?: (this: HTMLElement, ev: MouseEvent) => any): HTMLAnchorElement {
```

The question mark marks the parameter as optional. `onClick` will now have the type `(this: HTMLElement, ev: MouseEvent) => any | undefined`.

Then, when you add then event listener, just add a check for it:

```ts
if (onClick) {
    i.addEventLIstener("click", onClick);
}
```
Inside the `if (onClick)` block the type of onClick is just `(this: HTMLElement, ev: MouseEvent) => any`, because TypeScript is able to infer that it cannot be undefined at that point, so this works.

Other than that, LGTM.